### PR TITLE
feat(eslint-plugin): [unbound-method] improve error message

### LINKED ISSUE #3201: [unbound-method] `this: void` syntax should be highlighted in document and suggested in message
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

https://github.com/sveltejs/eslint-plugin-svelte3/issues/103#issuecomment-802415147

```json
{
  "rules": {
    "@typescript-eslint/unbound-method": 2
  }
}
```

```ts
// on:animationend={myStore.action}

const myStore = {
  action(event: AnimationEvent | string) {
    // handle event here
  },
}
```

**Expected Result**

Better message and better document for suggesting `this: void`

**Actual Result**

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.10.0` |
| `@typescript-eslint/parser`        | `4.10.0` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `14.16.0` |

### PR DESCRIPTION
**Pull request by JounQin**
close #3201

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/unbound-method.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Warns when a method is used outside of a method call.
 
 Class functions don't preserve the class scope when passed as standalone variables.
 
+If you think it is intended, you can try to add `this: void` as the first function parameter or use arrow function instead.
```
Comment (bradzacher) [inline packages/eslint-plugin/docs/rules/unbound-method.md:7]:
```suggestion
If your function does not access `this`, [you can annotate it with `this: void`](https://www.typescriptlang.org/docs/handbook/2/functions.html#declaring-this-in-a-function), or consider using an arrow function instead.
```

File: packages/eslint-plugin/src/rules/unbound-method.ts [hunk 2]
```diff
@@ -132,8 +132,8 @@ export default util.createRule<Options, MessageIds>({
       requiresTypeChecking: true,
     },
     messages: {
-      unbound:
-        'Avoid referencing unbound methods which may cause unintentional scoping of `this`.',
+      unbound: `Avoid referencing unbound methods which may cause unintentional scoping of \`this\`.
+If you think it is intended, you can try to add \`this: void\` as the first function parameter or use arrow function instead.`,
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/unbound-method.ts:136]: I don't think that we should have this in the message unless we know that the function does not have a `this` parameter.

eg it'd be weird if the user saw this message but the function was annotated with `this: Something`.

**Review by bradzacher — CHANGES_REQUESTED** [review]
thanks! a few comments

File: packages/eslint-plugin/src/rules/unbound-method.ts (hunk 2 above)
Comment (JounQin) [reply packages/eslint-plugin/src/rules/unbound-method.ts:136]: OK, I'll add a check and new message for it.

Comment (JounQin) [reply packages/eslint-plugin/src/rules/unbound-method.ts]: Done!

**Review by bradzacher — APPROVED** [review]
LGTM - thanks for adding this!

**Pull request merged**